tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java

```diff
@@ -2516,4 +2456,42 @@ public boolean canReadAndWrite() {
                 return false;
         }
     }
+
+    private static UnicodeMap<PageId> nonEmojiMap = null;
+
+    /**
+     * Return the PageId for the given character
+     *
+     * @param cp the character as a string
+     * @return the PageId
+     */
+    private static PageId getCharacterPageId(String cp) {
+        if (Emoji.getAllRgiNoES().contains(cp)) {
+            return Emoji.getPageId(cp);
+        }
+        if (nonEmojiMap == null) {
+            nonEmojiMap = createNonEmojiMap();
+        }
+        PageId pageId = nonEmojiMap.get(cp);
+        if (pageId == null) {
+            throw new InternalCldrException("Failure getting character page id");
+        }
+        // TODO: enable splitting Volume into Volume/Volume2, starting Volume2 at "acre-foot"
+        // Reference: https://unicode-org.atlassian.net/browse/CLDR-11155
+        return pageId;
+    }
+
+    /**
+     * Create the map from non-emoji characters to pages. Call with lazy initialization to avoid
+     * static initialization bugs, otherwise PageId.OtherSymbols could still be null.
```

Comment:
When nonEmojiMap was static initialized, surprisingly PageId.OtherSymbols was still null, and that's why I was getting strange results earlier -- putAll was putting null for PageId.OtherSymbols!